app/controllers/tags_controller.rb

```diff
@@ -90,23 +88,34 @@ def show
   end
 
   def feed
-    begin
-      @tag = Tag.find(params[:id])
-    rescue ActiveRecord::RecordNotFound
-       raise ActiveRecord::RecordNotFound, "Couldn't find tag with id '#{params[:id]}'"
-    end
-    if !@tag.canonical? && @tag.merger
-      @tag = @tag.merger
-    end
-    # Temp for testing
-    if %w(Fandom Character Relationship).include?(@tag.type.to_s) || @tag.name == "F/F"
-      if @tag.canonical?
-        @works = @tag.filtered_works.visible_to_all.order("created_at DESC").limit(25)
+    hash = Rails.cache.fetch(Tag.tag_feeds_key(params[:id])) do
+      path = nil
+      works = nil 
+      begin
+        tag = Tag.find(params[:id])
+      rescue ActiveRecord::RecordNotFound
+        raise ActiveRecord::RecordNotFound, "Couldn't find tag with id '#{params[:id]}'"
+      end
+      if !tag.canonical? && tag.merger
+        tag = tag.merger
+      end
+      # Temp for testing
+      if %w(Fandom Character Relationship).include?(tag.type.to_s) || tag.name == "F/F"
+        if tag.canonical?
+          works = tag.filtered_works.visible_to_all.order("created_at DESC").limit(25).all
+        else
+          works = tag.works.visible_to_all.order("created_at DESC").limit(25).all
+        end
       else
-        @works = @tag.works.visible_to_all.order("created_at DESC").limit(25)
+        path = tag_works_path(tag_id: tag.to_param)
       end
-    else
-      redirect_to tag_works_path(:tag_id => @tag.to_param) and return
+      { path: path, works: works, tag: tag }
+    end
+    redirect = hash[:path]
+    @works = hash[:works]
+    @tag = hash[:tag]
+    unless redirect.nil? 
+      redirect_to path and return
     end
```

Comment:
You would need to test for this specifically with a feed that causes a redirect and it has to be actually cached so it has to be the second time the feed is fetched; are you doing so anywhere? Because if you just get the cached value of hash, the code inside should NOT be re-executed in which case the "path" variable never gets set and redirect_to path should fail. (Actually I would guess the redirect should fail anyway because path is a local variable inside the hash block?)